### ISSUE TITLE
Throws a lot of errors in node 6 environment

### ISSUE THREAD

**Issue by puranjayjain** [js-cli/js-v8flags#32]
The package throws a bunch of errors as mentioned below in node v6+

These are just the errors relevant to this package (feel free to ask for the entire logs)

==== JS stack trace =========================================

Security context: 00000326171C9E79 #0#
  118: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:108] [pc=000001D179F13465] (this=00000326171E5A91 #5#,openErr=0000032617104101 ,result=000001596B30AA91 #60#)
  119: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:36] [pc=000001D179F13323] (this=00000326171E5A91 #5#,err=0000032617104101 ,fd=000001596B30AA91 #60#)
  120: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:47] [pc=000001D179F130A0] (this=00000326171E5A91 #5#)
  118: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:108] [pc=000001D179F13465] (this=00000326171E5A91 #5#,openErr=0000032617104101 ,result=000001596B30AA91 #60#)
  119: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:36] [pc=000001D179F13323] (this=00000326171E5A91 #5#,err=0000032617104101 ,fd=000001596B30AA91 #60#)
  120: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:47] [pc=000001D179F130A0] (this=00000326171E5A91 #5#)
  118: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:108] [pc=000001D179F13465] (this=00000326171E5A91 #5#,openErr=0000032617104101 ,result=000001596B30AA91 #60#)
  119: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:36] [pc=000001D179F13323] (this=00000326171E5A91 #5#,err=0000032617104101 ,fd=000001596B30AA91 #60#)
  120: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:47] [pc=000001D179F130A0] (this=00000326171E5A91 #5#)

(node) v8::FunctionTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 00000326171C9E79 #0#
  118: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:108] [pc=000001D179F13465] (this=00000326171E5A91 #5#,openErr=0000032617104101 ,result=000001596B30AA91 #60#)
  119: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:36] [pc=000001D179F13323] (this=00000326171E5A91 #5#,err=0000032617104101 ,fd=000001596B30AA91 #60#)
  120: /* anonymous */ [C:\Users\Puranjay\AppData\Roaming\npm\node_modules\gulp\node_modules\v8flags\index.js:47] [pc=000001D179F130A0] (this=00000326171E5A91 #5#)

=====================

**Comment by puranjayjain** [comment]
Node right now has documented a proper set of v8 flags, this module can still be supported but we can point instead point to node's apis for v6+ users. Please respond as to what path should be taken. I'll be happy to start a PR :)

**Comment by tkellen** [comment]
@puranjayjain Let's turn this library into a ponyfill. If we're on node 6x+ and we can return the exact same format we're returning for lower runtime versions, just bypass everything this lib is doing and do that (needs to be async tho).

**Comment by puranjayjain** [comment]
that's actually a really nice idea. how can I help?

**Comment by tkellen** [comment]
add another clause here to detect node runtimes that provide a list of flags and bail early:
https://github.com/js-cli/js-v8flags/blob/master/index.js#L99

**Comment by puranjayjain** [comment]
This would introduce breaking changes right? so we will be delivering it in the next major release right?

Also we need the checks in these exports only right?
https://github.com/js-cli/js-v8flags/blob/master/index.js#L96
https://github.com/js-cli/js-v8flags/blob/master/index.js#L106

**Comment by tkellen** [comment]
It shouldn't be a breaking change. As long as we call back on nextTick with the same response format we're using on older versions of node it should be transparent. The second line you've linked to should never be reached if we already have access to the flags.

**Comment by puranjayjain** [comment]
So essentially it will be a redesign of this module from scratch but keeping the apis same right? If that's the case it sounds fun :dancer:

**Comment by tkellen** [comment]
I mean, no, it should just be like
```js
  if (flagsAvailableByRuntime) {
    return process.nextTick(function () {
      cb(null, process.v8.flags);
    });
  }
```

...where `process.v8.flags` is whatever you can access the flags from in node 6+.

**Comment by puranjayjain** [comment]
Update: I now updated to the newer node release i.e. 6.2.0 and tested it again individually in: [x] marked are the ones where it ran flawlessly
* electron [x]
* node [x] on a gulp (3.9.1 based setup)
* node [] on a gulp (4.0#alpha2 based setup)

Where in the one in which it is failing (tried it on two different configurations) it is running without errors and in the other one it has the error. I investigated further on these lines I inferred that the error was due to too many logs being dumped and also due to memory leaks in this or another program. Really :disappointed: I wasted a lot of yours as well as my time on this issue. Feel free to close it for now.
